### DISCUSSION TITLE
Is there a way to save the last X searches so that if they are searched again it is faster the next time?

### DISCUSSION THREAD

**Discussion by tgkuus** [msiemens/tinydb#522] [Q&A]
I could be mistaken but it seems tinydb remembers the last search or maybe returns the same result if nothing about the query or database has changed.

I was wondering if there was a way to set the number of searches "remembered". Between 10 - 20 would speed up my application very significantly as it is the same 10-20 searches that happen in random order in a relatively large database (2000 docs or so). I realize I can save results, maybe in another table or in memory or something, however I wanted to know if there was a built in feature. Even more ideally if there was a setting that let me put in an arbitrary value for searches remembered.

**Comment by msiemens** [comment]
Hey! I think you're talking about the query cache feature. The documentation describes [how to adjust the size of the query cache](https://tinydb.readthedocs.io/en/latest/usage.html#query-caching):

> TinyDB caches query result for performance. That way re-running a query won’t have to read the data from the storage as long as the database hasn’t been modified. You can optimize the query cache size by passing the cache_size to the table(...) function:
> ```
> >>> table = db.table('table_name', cache_size=30)
> ```

Does that help you?
